internal/types/timestamp.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+type (
+	// Timestamp represents BSON type Timestamp.
+	Timestamp uint64
+)
+
+// timestampOperationCounter is an ordinal number for timestamps in the system.
```

Comment:
Can we explain in this comment why is this variable needed? I see it's increased every time when `NextTimestamp` is called. But why? I guess, when the app starts, this var is initialized with `0` (I don't see any special initialization)... And then every time when `NextTimestamp` is called, this var increases. What is the use case for it?

Reply:
> Can we explain in this comment why is this variable needed? I see it's increased every time when `NextTimestamp` is called. But why? I guess, when the app starts, this var is initialized with `0` (I don't see any special initialization)... And then every time when `NextTimestamp` is called, this var increases. What is the use case for it?

It is to differentiate the timestamps between operations. Actually, it can be not for the entire system but inside a single one timestamp. It depends on our chosen way, @AlekSi WDYT?

Reply:
It must be process-wide if we ever implement MongoDB-like replication. And it does not hurt to be process-wide even if we don't.

It and some other arguments, however, should be uint32, not uint64.

And let's use some shorter names. For example, we use `objectIDCounter`, so that one can be `timestampCounter` and argument names could be `c` and `ts`.

Reply:
Another option, that is a bit closer to https://bsonspec.org/spec.html: `timestampIncrement` and `inc`

Reply:
> And let's use some shorter names. For example, we use `objectIDCounter`, so that one can be `timestampCounter` and argument names could be `c` and `ts`.

variable renamed to `timestampCounter`.
variable name `ts` looks understandable.
variable name `c` looks not clear. It's not `sec` since it's `c`. What is it?

Reply:
ah, `c` - counter!

Reply:
> It and some other arguments, however, should be uint32, not uint64.

`timestampCounter` type changed to uint32.
In other places int64 for timestamp as an original type.